### PR TITLE
Feature/248 geoblazor templates

### LINKED ISSUE #248: Create GeoBlazor Templates
- Create template projects for the following
  - Blazor Web App (.NET 8)
  - Blazor Server
  - Blazor WebAssembly
  - Blazor MAUI Hybrid
- Investigate the best way to deploy these templates to customers

### PR DESCRIPTION
**Pull request by AndersenBell**
closes #248 

Adding a template project to house the starter templates for GeoBlazor.

### CODE REVIEW

File: templates/README.md
Comment (seahro) [inline templates/README.md:12]: i am excited to try these templates

File: templates/content/GeoBlazor.Template.Maui/Pages/Index.razor
Comment (seahro) [inline templates/content/GeoBlazor.Template.Maui/Pages/Index.razor:9]: I cant remember if you need an api key to fire these maps up the first time...

File: templates/content/GeoBlazor.Template.WebApp/GeoBlazor.Template.WebApp.Client/Pages/Counter.razor
Comment (seahro) [inline templates/content/GeoBlazor.Template.WebApp/GeoBlazor.Template.WebApp.Client/Pages/Counter.razor:12]: Even the counter pages get a map 👍

**Review by seahro — APPROVED** [review]
Looks awesome!  I am excited to take them for a test drive. I guess these files are auto generated: weatherforecast.cs, weatherforecastService.cs, Counter.razor, FetchData.razor, weather.json? Good to drop the base maps on various razor pages.  Probably for a future iteration, I wonder if in place of the weather.json file there can be some preset points or other data that can be built into an initial test layer data for the templates?

File: templates/content/GeoBlazor.Template.Maui/Resources/Splash/splash.svg
Comment (TimPurdum) [inline templates/content/GeoBlazor.Template.Maui/Resources/Splash/splash.svg:8]: Should we replace the image assets with GeoBlazor ones? Might be cool to "theme" the defaults.

File: templates/content/GeoBlazor.Template.Server/GeoBlazor.Template.Server.csproj
Comment (TimPurdum) [inline templates/content/GeoBlazor.Template.Server/GeoBlazor.Template.Server.csproj:7]: I know we started talking about this yesterday. I think it's a bad idea for us to set the `UserSecretsId` for a template. We should instead put the blank `ArcGISApiKey` setting in `appsettings.json` or `appsettings.Development.json` like in the Maui template, and add instructions to the ReadMe about using user secrets instead.

Here's a scenario why this matters. A user who has multiple apps started with this template would end up with shared user secrets, which they might not want.

File: templates/content/GeoBlazor.Template.Server/GeoBlazor.Template.Server.sln
Comment (TimPurdum) [inline templates/content/GeoBlazor.Template.Server/GeoBlazor.Template.Server.sln:2]: why is there a solution file at this level?

File: templates/content/GeoBlazor.Template.WebApp/GeoBlazor.Template.WebApp/Components/App.razor
Comment (TimPurdum) [inline templates/content/GeoBlazor.Template.WebApp/GeoBlazor.Template.WebApp/Components/App.razor:17]: Adding `@rendermode=...` here I believe will make the app work.

Comment (TimPurdum) [reply templates/content/GeoBlazor.Template.WebApp/GeoBlazor.Template.WebApp/Components/App.razor:17]: Or, probably better for a template, create a wrapper component where you can set the render mode for just the mapview, and add some comments there. See the .NET 8 branch for more info.

File: templates/content/GeoBlazor.Template.WebApp/GeoBlazor.Template.WebApp/Components/Layout/MainLayout.razor
Comment (TimPurdum) [inline templates/content/GeoBlazor.Template.WebApp/GeoBlazor.Template.WebApp/Components/Layout/MainLayout.razor:1]: I'm confused why we have this `Components` folder outside of the two project folders?

File: templates/content/GeoBlazor.Template.WebApp/GeoBlazor.Template.WebApp/GeoBlazor.Template.WebApp.csproj
Comment (TimPurdum) [inline templates/content/GeoBlazor.Template.WebApp/GeoBlazor.Template.WebApp/GeoBlazor.Template.WebApp.csproj:1]: There seems to be two whole copies of `GeoBlazor.Template.WebApp`, one inside the other one.

File: templates/README.md
Comment (TimPurdum) [inline templates/README.md:19]: suggested naming improvements for templates (shorter and easier to read):
- geoblazor-server
- geoblazor-wasm
- geoblazor-maui

File: templates/content/GeoBlazor.Template.WebAssembly/README.md
Comment (TimPurdum) [inline templates/content/GeoBlazor.Template.WebAssembly/README.md:11]: I feel like there might be other patterns in GB that use the `IConfiguration`?

**Review by TimPurdum — CHANGES_REQUESTED** [review]
Nice work! I'd like to see one more iteration, but these look great.

File: templates/content/GeoBlazor.Template.Maui/Pages/Index.razor
Comment (AndersenBell) [reply templates/content/GeoBlazor.Template.Maui/Pages/Index.razor:9]: you do, the readme.md contains where to set the apikey

File: templates/content/GeoBlazor.Template.Maui/Pages/FetchData.razor
Comment (AndersenBell) [reply templates/content/GeoBlazor.Template.Maui/Pages/FetchData.razor:1]: I like that this is just the default template with everything as if it was created from the Microsoft templates, we just added maps

File: templates/content/GeoBlazor.Template.Maui/Resources/Splash/splash.svg
Comment (AndersenBell) [reply templates/content/GeoBlazor.Template.Maui/Resources/Splash/splash.svg:8]: I am not opposed to replacing them

File: templates/content/GeoBlazor.Template.Server/GeoBlazor.Template.Server.sln
Comment (AndersenBell) [reply templates/content/GeoBlazor.Template.Server/GeoBlazor.Template.Server.sln:2]: it was just convenient for building/testing, I can remove it

File: templates/content/GeoBlazor.Template.WebAssembly/README.md
Comment (AndersenBell) [reply templates/content/GeoBlazor.Template.WebAssembly/README.md:11]: right now its the only one. For WebAssembly, it was not added by default, and right now it only reads the ArcGisToken

File: templates/README.md
Comment (AndersenBell) [reply templates/README.md:19]: I updated them, but I liked template in the name to not confuse them with the default package

**Comment by morehavoc** [comment]
I just came here to say that I'm excited about getting templates!

**Review by TimPurdum — APPROVED** [review]
I think these are mostly good. I want to actually test them and probably merge in .NET 8 today before we merge this in, so hold off on merging please.

**Pull request merged**
